### ISSUE TITLE
Codebase Completeness

### ISSUE THREAD

**Issue by jonasbbin** [madaan/self-refine#9]
Great work! We are currently trying to reproduce your results such that we can build on top of your insights. I see that you are still working on this codebase. Are there already some benchmarks that are fully implemented in this repository that you do not intend to develop further in the near future? Because when I am running the different benchmarks I sometimes run into some errors and I am not sure when the error is from my side, or just because some functionality is missing.

For example: When I am running the CommonGen benchmark on a reduced test set, I observe a lot of errors in the output file. My main source of error seems that the feedback from GPT does not have the intended structure, such that exceptions in the code occur. I did not change any of the training prompts/instructions prompts you provide in this file and use "gpt-3.5-turbo". Did you also observe this behaviour?

I also noticed two different things, which I thought I would notify you about:

- Compared to other benchmarks you do not include the refinement history in the GSM-8k benchmark. You only use the training prompt, the instruction prompt and the current iteration. Is this intended?
- The run.py file is missing for the sentiment_reversal benchmark and I can't find the training data for the Code Readability Improvement

Thanks a lot!

**Comment by madaan** [comment]
Hey,

Thanks for pointing out the issues, and sorry about the missing files. We have a private repo where some of the development happens, and it went out of sync. I've added these files.

The GSM with prompting being updated is now located [here](https://github.com/madaan/self-refine/blob/c8a75a2e945598b707ce4a2b6f14e51235bf82ee/src/gsm/feedback.py#L56).

Re errors, unfortunately some of it is due to the inability of gpt-3.5-turbo to faithfully follow the prompt format. Please see the discussion [here](https://github.com/madaan/self-refine/issues/6) for more.

**Comment by jonasbbin** [comment]
Great Thanks!
